### PR TITLE
Fix some issues reported by Coverity scan

### PR DESCRIPTION
**Pull request by frankmorgner**
##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

**Comment by frankmorgner** [comment]
@popovec , could you please verify the fix in card-myeid.c?

File: src/smm/sm-cwa14890.c [hunk 1]
```diff
@@ -139,6 +139,8 @@ sm_cwa_decode_authentication_data(struct sc_context *ctx, struct sm_cwa_keyset *
 		LOG_FUNC_RETURN(ctx, SC_ERROR_SM_AUTHENTICATION_FAILED);
 
 	rv = sm_decrypt_des_cbc3(ctx, keyset->enc, session_data->mdata, session_data->mdata_len, &decrypted, &decrypted_len);
+	if (rv != SC_SUCCESS)
+		free(decrypted);
```
Comment (Jakuje) [inline src/smm/sm-cwa14890.c:143]: would not be better if the `sm_decrypt_des_cbc3()` would not allocate/free the output parameter if it fails? With this you will have to fix the same issue in `sm-card-iasecc.c`.

Comment (frankmorgner) [reply src/smm/sm-cwa14890.c:143]: I didn't want to refactor this too much, but I can try to.

Comment (frankmorgner) [reply src/smm/sm-cwa14890.c]: done

**Review by Jakuje — APPROVED** [review]

**Pull request merged**